signing/src/test/java/tech/pegasys/web3signer/signing/config/SignerLoaderEnhancedTest.java

```diff
@@ -0,0 +1,486 @@
+        SignerLoader.builder()
+            .configsDirectory(configsDirectory)
+            .parallelProcess(true)
+            .batchSize(100)
+            .build();
+    assertThat(signerLoader.getBatchSize()).isEqualTo(100);
+
+    signerLoader.load(signerParser);
+
+    // Should see parallel processing but not batch-specific messages
+    ExpectedLoggingAssertions.assertThat(logging)
+        .hasInfoMessage("Processing 100 files in parallel with batch size 100");
+
+    // Verify no batch subdivision occurred by checking all log messages
+    assertThat(
+            logging.getLogEvents().stream()
+                .map(LogEvent::getMessage)
+                .noneMatch(msg -> msg.contains("Processing batch")))
+        .isTrue();
+  }
+
+  @Test
+  void batchProgressOnlyLoggedWhenTotalExceedsBatchSize() throws Exception {
+    int fileCount = 150;
+    createBLSRawConfigFiles(fileCount);
+
+    signerLoader =
+        SignerLoader.builder()
+            .configsDirectory(configsDirectory)
+            .parallelProcess(true)
+            .batchSize(100)
+            .build();
+    assertThat(signerLoader.getBatchSize()).isEqualTo(100);
+
+    signerLoader.load(signerParser);
+
+    // With 150 files and batch size 100, should see 2 batches
+    ExpectedLoggingAssertions.assertThat(logging)
+        .hasInfoMessage("Processing 150 files in parallel with batch size 100")
+        .hasInfoMessage("Processing batch 1-100 of 150 files")
+        .hasInfoMessage("Processing batch 101-150 of 150 files");
+  }
+
+  // ==================== SEQUENTIAL VS PARALLEL PROCESSING TESTS ====================
+
+  @Test
+  void sequentialProcessingWhenBelowThreshold() throws Exception {
+    // Create files below sequential threshold (default 100)
+    createBLSRawConfigFiles(50);
+
+    signerLoader =
+        SignerLoader.builder()
+            .configsDirectory(configsDirectory)
+            .parallelProcess(true)
+            .sequentialThreshold(100)
+            .build();
+
+    MappedResults<ArtifactSigner> result = signerLoader.load(signerParser);
+
+    assertThat(result.getValues()).hasSize(50);
+    assertThat(result.getErrorCount()).isZero();
+
+    // Verify sequential processing was used
+    ExpectedLoggingAssertions.assertThat(logging)
+        .hasInfoMessage("Processing 50 files sequentially");
+  }
+
+  @Test
+  void parallelProcessingWhenAboveThreshold() throws Exception {
+    // Create files above sequential threshold
+    createBLSRawConfigFiles(150);
+
+    signerLoader =
+        SignerLoader.builder()
+            .configsDirectory(configsDirectory)
+            .parallelProcess(true)
+            .sequentialThreshold(100)
+            .build();
+
+    final MappedResults<ArtifactSigner> result = signerLoader.load(signerParser);
+
+    assertThat(result.getValues()).hasSize(150);
+    assertThat(result.getErrorCount()).isZero();
+
+    // Verify parallel processing was used
+    ExpectedLoggingAssertions.assertThat(logging)
```

Comment:
Similar to above test cant this also be updated to verify that signers were loaded in parallel? This is only checking the logs are outputting correctly.

Reply:
Updated test cases to use Mockito verify to verify method invocations.